distinct.go

```diff
@@ -0,0 +1,50 @@
+package gcf
+
+type distinctIterable[T comparable] struct {
+	itb Iterable[T]
+}
+
+type distinctIterator[T comparable] struct {
+	it      Iterator[T]
+	past    map[T]struct{}
+	current T
+}
+
+// Distinct makes Iterable contains unique elements.
```

Comment:
```suggestion
// Distinct makes Iterable contains unique elements.
// Inner elements is restrict by comparable constraint.
```

大半のIterableがany要素の対象なので、これ書いておかないと使えないケースが分かりづらい気がする